### PR TITLE
Organized Vagrant folder 

### PR DESCRIPTION
**Pull request by OneCricketeer**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,5 +1,7 @@
 twitter4j.properties
 vagrant/rpms/
+vagrant/deps/
+vagrant_deps/
```
Comment (risdenk) [inline .gitignore]: Why two new ignores?

File: Vagrantfile [hunk 2]
```diff
@@ -35,20 +39,53 @@ Vagrant.configure(2) do |config|
 
   config.vm.synced_folder ".", "/vagrant", type: "nfs"
 
+  domain_suffix = "demo"
   machines.each do |machine|
-    config.vm.define machine[:name] do |vmconfig|
-      vmconfig.vm.box = "bento/centos-6.7"
+    short_name = machine[:name]
+    domain_name = "#{short_name}.#{domain_suffix}"
```
Comment (risdenk) [inline Vagrantfile:45]: why a variable? can't this be inlined on line 55?

File: Vagrantfile [hunk 3]
```diff
@@ -35,20 +39,53 @@ Vagrant.configure(2) do |config|
...
+      vmconfig.vm.hostname = domain_name
       vmconfig.vm.network :private_network, ip: machine[:ip]
 
       vmconfig.vm.provision "ansible" do |ansible|
-        ansible.playbook = machine[:ansible]
+        ansible.playbook = ANSIBLE_PLAYBOOK
+        ansible.verbose = 'vvvv'
+
+        # Use helper method to define machine groups
+        ansible.groups = group_up(machines)
```
Comment (risdenk) [inline Vagrantfile:61]: is this really necessary? can't groups be defined some other way?

File: Vagrantfile [hunk 4]
```diff
@@ -35,20 +39,53 @@ Vagrant.configure(2) do |config|
...
+      end
+    end
+  end
+end
+
+# Helper method to define machine groupings
+def group_up (machines)
```
Comment (risdenk) [inline Vagrantfile:84]: did you right this yourself or copy form somewhere else online?

File: ansible.cfg [hunk 5]
```diff
@@ -0,0 +1,3 @@
+[defaults]
+
+roles_path = provisioning/roles
```
Comment (risdenk) [inline ansible.cfg:3]: does this file only get applied if running ansible from top level dir or does it apply whereever you run ansible from?

File: .gitignore (hunk 1 above)
Comment (OneCricketeer) [inline .gitignore]: Forgot about those. I deleted the vagrant folder.

File: Vagrantfile (hunk 2 above)
Comment (OneCricketeer) [inline Vagrantfile:45]: Can be inlined, sure.

**Comment by risdenk** [comment]
Why change vagrant host from A.demo to just A? Does this improve usability?

File: Vagrantfile (hunk 3 above)
Comment (OneCricketeer) [inline Vagrantfile:61]: Not that I know of.

File: Vagrantfile (hunk 4 above)
Comment (OneCricketeer) [inline Vagrantfile:84]: I wrote it.

File: Vagrantfile [hunk 6]
```diff
@@ -35,20 +39,53 @@ Vagrant.configure(2) do |config|
...
       end
 
-      vmconfig.vm.hostname = machine[:name]
+      vmconfig.vm.hostname = domain_name
       vmconfig.vm.network :private_network, ip: machine[:ip]
 
       vmconfig.vm.provision "ansible" do |ansible|
-        ansible.playbook = machine[:ansible]
+        ansible.playbook = ANSIBLE_PLAYBOOK
+        ansible.verbose = 'vvvv'
```
Comment (risdenk) [inline Vagrantfile]: should the verbosity be increased for this commit? or was this just for debugging?

File: ansible.cfg (hunk 5 above)
Comment (OneCricketeer) [inline ansible.cfg:3]: Ansible config applied in this order
- ANSIBLE_CONFIG (an environment variable)
- ansible.cfg (in the current directory)
- .ansible.cfg (in the home directory)
- /etc/ansible/ansible.cfg

Source: http://docs.ansible.com/ansible/intro_configuration.html

File: Vagrantfile (hunk 6 above)
Comment (OneCricketeer) [inline Vagrantfile]: Debugging only - making new commit.

**Comment by risdenk** [comment]
This PR doesn't just organize the vagrant folder it actually removes it...

File: ansible.cfg (hunk 5 above)
Comment (risdenk) [inline ansible.cfg:3]: yea guess the part I missed was that the whole vagrant directory was removed.

File: Vagrantfile (hunk 4 above)
Comment (risdenk) [inline Vagrantfile:84]: can you add a comment to describe what this method actually does? Why it is actually necessary?

**Comment by risdenk** [comment]
There are a lot of changes in this PR:
- removed vagrant folder altogether
- renamed A.demo to A for vagrant commands (leaves hostnames as A.demo)
- changes Ansible scripts

@cricket007 what testing has been done for these changes?
- multiple up/down cycles?
- testing dependency caching?
- testing repeated provisions are still idempotent?

**Comment by OneCricketeer** [comment]
@risdenk Testing done: 
- multiple up/down cycles with & without dependency caching
- repeated provisions for all machines except HDP are still idempotent

File: provisioning/roles/couchbase/defaults/main.yml [hunk 7]
```diff
@@ -0,0 +1 @@
+---
```
Comment (risdenk) [inline provisioning/roles/couchbase/defaults/main.yml]: is this file necessary?

File: provisioning/roles/common/handlers/main.yml [hunk 8]
```diff
@@ -0,0 +1,4 @@
+---
```
Comment (risdenk) [inline provisioning/roles/common/handlers/main.yml]: what is the purpose of this file?

File: provisioning/playbooks/nodejs/setup_elasticsearch-twitter-webapp.yml [hunk 9]
```diff
@@ -0,0 +1,18 @@
...
+    - name: copy app
+      command: cp -rf {{ nodejs_app_dir }} /opt/
+      args:
+        creates: /opt/{{ nodejs_app_name }}/
+
+    - name: install app
+      npm: path=/opt/{{ nodejs_app_name }}
+
+    - name: start the app forever
+      shell: forever start -c "npm start --unsafe-perm" /opt/{{ nodejs_app_name }}/
```
Comment (risdenk) [inline provisioning/playbooks/nodejs/setup_elasticsearch-twitter-webapp.yml]: does this need to be a shell or can command be used?

File: provisioning/roles/couchbase/defaults/main.yml (hunk 7 above)
Comment (OneCricketeer) [inline provisioning/roles/couchbase/defaults/main.yml]: I moved the variables in there to group_vars, so nope.

File: provisioning/roles/common/handlers/main.yml (hunk 8 above)
Comment (OneCricketeer) [inline provisioning/roles/common/handlers/main.yml]: It's used for `notify: Source local settings` in couchbase and hdp roles `tasks/main.yml` under Disable THP

File: provisioning/roles/couchbase/tasks/main.yml [hunk 10]
```diff
@@ -0,0 +1,29 @@
...
+# pip installs
+- name: Install httplib2
+  pip: name=httplib2
+
+# Disable Transparent Huge Pages (THP) [http://docs.couchbase.com/admin/admin/Install/rhel-installing.html]
+- name: Disable THP
+  sudo: True
+  shell: echo 'for i in /sys/kernel/mm/*transparent_hugepage/enabled; do echo never > $i; done' >> /etc/rc.local
+  shell: echo 'for i in /sys/kernel/mm/*transparent_hugepage/defrag; do echo never > $i; done' >> /etc/rc.local
+  shell: for i in /sys/kernel/mm/*transparent_hugepage/enabled; do echo never > $i; done
```
Comment (risdenk) [inline provisioning/roles/couchbase/tasks/main.yml]: should have one of these for /defrag as well.

File: provisioning/roles/couchbase/tasks/main.yml [hunk 11]
```diff
@@ -0,0 +1,29 @@
...
+  when: ansible_distribution_version != "12.04"
+  notify: Source local settings
```
Comment (risdenk) [inline provisioning/roles/couchbase/tasks/main.yml]: shouldn't need this with the echo never > enabled/defrag

File: provisioning/roles/common/handlers/main.yml (hunk 8 above)
Comment (risdenk) [inline provisioning/roles/common/handlers/main.yml]: shouldn't need this with the new comments for disable thp

File: provisioning/playbooks/couchbase/create_bucket.yml [hunk 12]
```diff
@@ -0,0 +1,34 @@
...
+  sudo: False
+  gather_facts: False
+  tasks:
+    - name: Create cluster grouping
+      group_by: key={{ couchbase_server_node_role }}
+
+- hosts: primary
+  sudo: False
+  gather_facts: False
+  vars:
```
Comment (risdenk) [inline provisioning/playbooks/couchbase/create_bucket.yml:12]: should these vars still be here? how is this different than the couchbase group vars?

File: provisioning/playbooks/couchbase/install_couchbase.yml [hunk 13]
```diff
@@ -0,0 +1,18 @@
...
+    - common
+    - couchbase
+  tasks:
+    - name: Couchbase Server Service
+      service: name=couchbase-server state=started enabled=yes
+
+    - name: Wait for Admin
+      wait_for: port=8091 delay=20
+
+    - name: Wait for EPMD
```
Comment (risdenk) [inline provisioning/playbooks/couchbase/install_couchbase.yml]: what is EPMD?

File: provisioning/roles/couchbase/tasks/main.yml [hunk 14]
```diff
@@ -0,0 +1,30 @@
...
+
+# Disable Transparent Huge Pages (THP) [http://docs.couchbase.com/admin/admin/Install/rhel-installing.html]
+- name: Disable THP
+  sudo: True
+  shell: echo 'for i in /sys/kernel/mm/*transparent_hugepage/enabled; do echo never > $i; done' >> /etc/rc.local
+  shell: echo 'for i in /sys/kernel/mm/*transparent_hugepage/defrag; do echo never > $i; done' >> /etc/rc.local
+  shell: for i in /sys/kernel/mm/*transparent_hugepage/enabled; do echo never > $i; done
+  shell: for i in /sys/kernel/mm/*transparent_hugepage/defrag; do echo never > $i; done
+  when: ansible_distribution_version != "12.04"
+  notify: Source local settings
```
Comment (risdenk) [inline provisioning/roles/couchbase/tasks/main.yml]: Does this notify and related handler need to still exist?

File: provisioning/roles/elastic/tasks/main.yml [hunk 15]
```diff
@@ -0,0 +1,14 @@
+---
+
+- name: Distribution specific variables
+  include_vars: "../defaults/{{ ansible_distribution }}.yml"
+
+# CentOS tasks
+- include: CentOS.yml
+  when: ansible_distribution  == "CentOS"
```
Comment (risdenk) [inline provisioning/roles/elastic/tasks/main.yml]: extra space

**Comment by risdenk** [comment]
@cricket007 
`vagrant/README.md` was removed but the contents were not added anywhere?

**Comment by risdenk** [comment]
@kruthar @westeras - Do you have any issues w/ the `vagrant` folder going away and `Vagrantfile` becoming top level? Easiest way to see this is here: https://github.com/cricket007/hdp-es-cb-demo/tree/vagrant-organize

**Comment by risdenk** [comment]
@cricket007 - Can `ansible.cfg` stuff be moved inside the `Vagrantfile` or does it need to be separate?

File: provisioning/playbooks/couchbase/install_couchbase.yml (hunk 13 above)
Comment (OneCricketeer) [inline provisioning/playbooks/couchbase/install_couchbase.yml]: Couchbase uses Erlang to communicate between nodes of a cluster. EPMD means Erlang Port Mapper Daemon. Could remove this bit because this is a single node Couchbase cluster.

Comment (risdenk) [inline provisioning/playbooks/couchbase/install_couchbase.yml]: Can this note be added to the code? Also if its not needed where did it come from?

**Comment by OneCricketeer** [comment]
@risdenk `ansible.cfg` has to be a file in order to override the [ansible default variables](https://raw.githubusercontent.com/ansible/ansible/devel/examples/ansible.cfg)

File: provisioning/playbooks/couchbase/create_bucket.yml (hunk 12 above)
Comment (OneCricketeer) [inline provisioning/playbooks/couchbase/create_bucket.yml:12]: These are variables specific for the bucket creation, not the entire group of machines.

File: provisioning/roles/elastic/tasks/main.yml (hunk 15 above)
Comment (OneCricketeer) [inline provisioning/roles/elastic/tasks/main.yml]: really?

Comment (risdenk) [inline provisioning/roles/elastic/tasks/main.yml]: yes before the ==

File: provisioning/playbooks/couchbase/create_bucket.yml (hunk 12 above)
Comment (risdenk) [inline provisioning/playbooks/couchbase/create_bucket.yml:12]: The variables are definitely related to the server and not just for the bucket.
bucket_ram is related to server_ram.
bucket_name is related to default_bucket.

**Comment by OneCricketeer** [comment]
@risdenk The contents of `vagrant/README.md` is included in the Setting up the Environment section of the root README

File: provisioning/roles/common/tasks/main.yml [hunk 16]
```diff
@@ -0,0 +1,5 @@
+---
+
+# CentOS tasks
+- include: CentOS.yml
+  when: ansible_distribution  == "CentOS"
```
Comment (risdenk) [inline provisioning/roles/common/tasks/main.yml]: Extra space before == again

File: provisioning/playbooks/couchbase/create_bucket.yml (hunk 12 above)
Comment (OneCricketeer) [inline provisioning/playbooks/couchbase/create_bucket.yml:12]: Yes, `bucket_ram <= server_ram`, but `bucket_name` is not related to `default_bucket`. `default_bucket` wasn't used, so it is being removed. This file creates a single bucket with the variables mentioned specifically for this bucket.

File: provisioning/roles/couchbase/tasks/main.yml [hunk 17]
```diff
@@ -0,0 +1,30 @@
...
+  stat: path=/vagrant/vagrant_deps/{{ couchbase_package }}
+  register: couchbase_local_package
+
+- name: Download Couchbase RPM locally
+  get_url: url={{ couchbase_url }} dest=/vagrant/vagrant_deps/{{ couchbase_package }}
+  when: not couchbase_local_package.stat.exists
+
+# CentOS tasks
+- include: CentOS.yml
+  when: ansible_distribution  == "CentOS"
```
Comment (risdenk) [inline provisioning/roles/couchbase/tasks/main.yml]: extra space before == again

File: provisioning/roles/hdp/tasks/main.yml [hunk 18]
```diff
@@ -0,0 +1,41 @@
+---
+
+# CentOS tasks
+- include: CentOS.yml
+  when: ansible_distribution  == "CentOS"
```
Comment (risdenk) [inline provisioning/roles/hdp/tasks/main.yml]: extra space before ==

File: provisioning/roles/nodejs/tasks/main.yml [hunk 19]
```diff
@@ -0,0 +1,9 @@
+---
+
+# CentOS tasks
+- include: CentOS.yml
+  when: ansible_distribution  == "CentOS"
```
Comment (risdenk) [inline provisioning/roles/nodejs/tasks/main.yml]: extra space before ==

File: provisioning/playbooks/couchbase/cluster_init.yml [hunk 20]
```diff
@@ -0,0 +1,29 @@
...
+- hosts: couchbase
+  sudo: False
+  gather_facts: False
+  tasks:
+    - name: Create cluster grouping
+      group_by: key={{ couchbase_server_node_role }}
+
+- hosts: primary
+  sudo: True
+#  serial: 1
```
Comment (risdenk) [inline provisioning/playbooks/couchbase/cluster_init.yml]: commented out? What did this do?

File: provisioning/playbooks/elastic/setup_cb_transport.yml [hunk 21]
```diff
@@ -0,0 +1,49 @@
+---
+- hosts: elastic
+  sudo: True
+  vars:
+    cb_transport_plugin_version: 2.0.0
```
Comment (risdenk) [inline provisioning/playbooks/elastic/setup_cb_transport.yml:5]: versions are pulled out for couchbase/es. Why not here?

File: provisioning/playbooks/elastic/setup_kibana.yml [hunk 22]
```diff
@@ -0,0 +1,41 @@
+---
+- hosts: elastic
+  sudo: True
+  vars:
+    kibana_version: 4.0.1
```
Comment (risdenk) [inline provisioning/playbooks/elastic/setup_kibana.yml:5]: versions are pulled out for couchbase/es. Why not here?

File: provisioning/playbooks/couchbase/cluster_init.yml (hunk 20 above)
Comment (OneCricketeer) [inline provisioning/playbooks/couchbase/cluster_init.yml]: I copied it from another task I had with multiple machines. 

> By default, Ansible will try to manage all of the machines referenced in a play in parallel ... you can define how many hosts Ansible should manage at a single time by using the ‘’serial’’ keyword
> http://docs.ansible.com/ansible/playbooks_delegation.html#rolling-update-batch-size

**Comment by risdenk** [comment]
We should parameterize the hdp ansible stuff as well for versions but since its not already pulled out thats going to be a pain. I'll create an issue for that specifically.

File: provisioning/playbooks/elastic/setup_cb_transport.yml (hunk 21 above)
Comment (OneCricketeer) [inline provisioning/playbooks/elastic/setup_cb_transport.yml:5]: Same reason as the couchbase create_bucket file, these variables are specific to setting up the Couchbase Transport plugin.

File: provisioning/playbooks/elastic/setup_kibana.yml (hunk 22 above)
Comment (OneCricketeer) [inline provisioning/playbooks/elastic/setup_kibana.yml:5]: Same reason as the couchbase create_bucket file, these variables are specific to setting up Kibana.

**Comment by risdenk** [comment]
@cricket007 Looking pretty good. Can you squash your commits?

File: provisioning/roles/nodejs/tasks/main.yml [hunk 23]
```diff
@@ -0,0 +1,9 @@
+---
+
+# CentOS tasks
+- include: CentOS.yml
+  when: ansible_distribution == "CentOS"
+
+- name: Install global npm packages
+  command: npm install -g {{ item }}
```
Comment (risdenk) [inline provisioning/roles/nodejs/tasks/main.yml]: any reason not to use the npm ansible module?

File: Vagrantfile [hunk 24]
```diff
@@ -0,0 +1,105 @@
...
+# # vi: set ft=ruby :
+
+ANSIBLE_PLAYBOOK = ENV['ANSIBLE_PLAYBOOK'] || "provisioning/site.yml"
+VAGRANTFILE_API_VERSION = 2
+BOX_NAME = ENV['BOX_NAME'] || "bento/centos-6.7"
+
+machines = [
+  {:name => "elasticsearch", :ip => "192.168.56.42", :memory => 1024, :groups => ["elastic"]},
+  {:name => "couchbase", :ip => "192.168.56.41", :memory => 1024, :groups => ["couchbase"]},
+  {:name => "hdp", :ip => "192.168.56.43", :memory => 4096, :groups => ["hdp"]},
```
Comment (risdenk) [inline Vagrantfile]: make sure this is commented out :)

**Comment by risdenk** [comment]
:+1:

**Comment by OneCricketeer** [comment]
:shipit:
